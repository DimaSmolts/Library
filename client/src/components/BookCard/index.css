.book-card {
  background: var(--gold);
  padding: 10px;
  box-sizing: border-box;
  border-radius: 15px;
  border: 2px solid var(--dark-blue);
}

.book-card:hover{
  box-shadow: 5px 5px 15px;
}

.book-card img {
  width: 100%;
  border-radius: 10px;
  box-sizing: border-box;
  border: 2px solid var(--dark-blue);
}

.book-card-link {
  text-decoration: none;
  color: var(--dark-blue);
}

.book-card-author {
  font-size: 0.7em;
}
