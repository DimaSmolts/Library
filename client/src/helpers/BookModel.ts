export class Book {
  id: number;
  name: string;
  author: string;
  genres: Array<string>;
  description: string;
  rate: number;
  price: number;
  image: string;
}
