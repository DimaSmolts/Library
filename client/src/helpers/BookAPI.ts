import axios from 'axios';
import { Book } from './BookModel'

const baseURL = process.env.REACT_APP_API_URL || 'http://localhost:3000';

const instance = axios.create({
  baseURL: baseURL + '/api',
})

export const bookAPI = {
  async getBooks() {
    return await instance.get<Array<Book>>('/books');
  },
  async getBook(id: number) {
    return await instance.get<Book>('/books/' + id);
  }
}
