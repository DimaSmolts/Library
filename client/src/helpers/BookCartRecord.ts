import { Book } from './BookModel'

export interface BookCartRecord {
  book: Book,
  quantity: number,
}
