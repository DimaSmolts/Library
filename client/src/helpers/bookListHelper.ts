import { Book } from './BookModel'

export function filterBookList(books: Array<Book>, query: string): Array<Book> {
  const temp = books.filter(book => {
    return book.name
      .trim()
      .toLowerCase()
      .includes(query.trim().toLowerCase())
  })
  return temp;
}
