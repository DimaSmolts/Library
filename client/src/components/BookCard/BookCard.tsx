import React from 'react'
import { Book } from '../../helpers/BookModel'
import { Link } from 'react-router-dom'
import './index.css'

interface BookCardProps {
  book: Book,
}

const BookCard: React.SFC<BookCardProps> = (props: BookCardProps) => {
  return <Link to={'/books/' + props.book.id} className='book-card-link' >
    <div className='book-card'>
      <img src={props.book.image} />
      <span className='book-card-author'>{props.book.author}</span>
      <h3>{props.book.name}</h3>
    </div>
  </Link>
}

export default BookCard
