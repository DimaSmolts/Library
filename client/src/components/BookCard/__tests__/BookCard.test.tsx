import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import BookCard from '../BookCard'
import { Book } from '../../../helpers/BookModel'

const book = { id: 1, name: 'name', image: 'image', author: 'author' } as Book

test('snapshot: <BookCard /> renders correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <BookCard book={book} />
    </BrowserRouter>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

describe('BookCard component', () => {
  it('should build link to BookDetails correctly', () => {
    const wrapper = mount(
      <BrowserRouter>
        <BookCard book={book} />
      </BrowserRouter>)

    expect(wrapper.find('a').prop('href')).toBe('/books/' + book.id)
  })

  it('should render book name and image src correctly', () => {
    const wrapper = mount(
      <BrowserRouter>
        <BookCard book={book} />
      </BrowserRouter>)

    expect(wrapper.find('h3').text()).toBe(book.name)
    expect(wrapper.find('img').prop('src')).toBe(book.image)
  })
})
