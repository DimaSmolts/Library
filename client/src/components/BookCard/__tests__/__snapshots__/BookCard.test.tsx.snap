// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: <BookCard /> renders correctly 1`] = `
<a
  className="book-card-link"
  href="/books/1"
  onClick={[Function]}
>
  <div
    className="book-card"
  >
    <img
      src="image"
    />
    <span
      className="book-card-author"
    >
      author
    </span>
    <h3>
      name
    </h3>
  </div>
</a>
`;
